jaq-interpret/src/mir.rs

```diff
@@ -38,9 +38,11 @@ pub enum Num {
 impl Num {
     fn parse(n: &str) -> Result<Self, Self> {
         if n.contains(['.', 'e', 'E']) {
-            n.parse::<f64>().map(Num::Float).map_err(|_| Num::Float(0.))
+            n.parse::<f64>()
+                .map(Num::Float)
+                .map_err(|_| Self::Float(0.))
         } else {
-            n.parse::<isize>().map(Num::Int).map_err(|_| Num::Int(0))
+            n.parse::<isize>().map(Num::Int).map_err(|_| Self::Int(0))
```

Comment:
It's a shame that this `Self` breaks the symmetry between the `if` and the `else` branch. Can we not get the `if` branch into a single line again?

Reply:
Fixed in https://github.com/01mf02/jaq/pull/124/commits/679f5f6689f1814f0fd85dca8fe2218628b1ccca